File: 프로그래머스/lv1/12947. 하샤드 수/하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    int result;
    if(x <= 10){
        return true;
    }
    else if(x > 10 && x < 100){
        if(x % (x/10 + x%10) == 0) return true;
        else return false;
    }
     else if(x >= 100 && x < 1000){
          if(x % (x/100 + (x%100)/10 + (x%100)%10) == 0) return true;
        else return false;
    }
     else if(x >= 1000 && x < 10000){
        if(x % (x/1000 + (x%1000)/100 + ((x%1000)%100)/10 + ((x%1000)%100)%10) == 0) return true;
        else return false;
    }
    else return true;
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
public class Solution {
    public boolean solution(int x) {
        int sum = 0;
        int originalX = x;
        
        while (x > 0) {
            sum += x % 10; // 숫자의 각 자릿수를 더함
            x /= 10; // 다음 자릿수로 이동
        }
        
        return originalX % sum == 0; // 원래 숫자를 자릿수의 합으로 나누어 떨어지는지 확인
    }
}
